docs/generate-time-series/python.rst

```diff
@@ -0,0 +1,201 @@
+`getloadavg`_ function:
+
+    >>>  psutil.getloadavg()
+    (4.59326171875, 4.56689453125, 5.150390625)
+
+The function returns a tuple with three `floats`_ corresponding to the one
+minute average, five-minute average, and 15-minute average.
+
+
+Set up CrateDB
+==============
+
+Make sure CrateDB running on ``localhost`` with the default configuration.
+
+Then, import the `crate`_ library:
+
+    >>>  from crate import client
+
+`Connect`_ to CrateDB:
+
+    >>>  connection = client.connect()
+
+Get a `cursor`_:
+
+    >>>  cursor = connection.cursor()
+
+And, finally, `create a table`_ suitable for writing load averages:
+
+    >>> cursor.execute(
+    ...     """CREATE TABLE load (
+    ...            timestamp TIMESTAMP GENERATED ALWAYS AS CURRENT_TIMESTAMP,
+    ...            avg_1m REAL,
+    ...            avg_5m REAL,
+    ...            avg_15m REAL)"""
+    ... )
+
+In the `CrateDB Admin UI`_, you should see the table you created after navigating
+to the *Tables* screen using the left-hand navigation menu:
+
+.. image:: ../_assets/img/generate-time-series/table.png
+
+
+Record your system load
+=======================
+
+Now you can start sampling the load averages and writing the results to
+CrateDB.
+
+The following command calls `getloadavg`_ and uses the result tuple as `input
+values`_ for the `INSERT`_ query:
+
+    >>> cursor.execute(
+    ...     "INSERT INTO load (avg_1m, avg_5m, avg_15m) VALUES (?, ?, ?)",
+    ...     psutil.getloadavg(),
+    ... )
+
+Press the up arrow on your keyboard and hit *Enter* to run the same command a
+few more times.
+
+When you're done, you can `SELECT`_ that data back out of CrateDB, like so:
+
+    >>> cursor.execute('SELECT * FROM load ORDER BY timestamp DESC')
+
+Then, `fetch all`_ the result rows at once:
+
+    >>> cursor.fetchall()
+    [[1580670546270, 5.4072266, 4.788086, 4.620117],
+     [1580670545367, 5.6171875, 4.8183594, 4.629883],
+     [1580670478666, 4.411621, 4.578125, 4.54541]]
+
+Cool, it worked! We recorded three sets of load averages with a corresponding
+timestamp.
+
+
+Automate it
+===========
+
+Now we have the basics figured out, let's automate the data collection.
+
+Collect the code you want to use into a single file. For example:
+
+.. code-block:: python
+
+    import time
+
+    import psutil
+    from crate import client
+
+
+    def insert():
+        # New connection each time
+        connection = client.connect()
+        print("CONNECT OK")
+        cursor = connection.cursor()
+        cursor.execute(
+            "INSERT INTO load (avg_1m, avg_5m, avg_15m) VALUES (?, ?, ?)",
+            psutil.getloadavg(),
+        )
+        print("INSERT OK")
+
+
+    # Loop indefinitely
+    while True:
+        insert()
+        print("Sleeping for 10 seconds...")
+        time.sleep(10)
+
+Here, the script sleeps for 10 seconds after each sample. Therefore, my time
+series data will have a *resolution* of 10 seconds. You may want to choose a
```

Comment:
Maybe it would be cool to use `argparse` or some other method of getting arguments in a script so that the user can play around with it in the command line, like `python monitor-load.py --sleep 5` or something?

Reply:
I agree this would be cool. but I am hoping that readers go on to play around with the file they just created by editing values, adding code, etc, etc

a previous version of this code was randomly sampling utilization stats from one of the available cpus so the stream of data was replicating a setup with multiple sensors all recording the same type of data. but ultimately I cut it because I wanted to pair down the code to the smallest possible thing that would work